Source/NSubstitute/SubstituteExtensions.cs

```diff
@@ -142,7 +142,7 @@ internal static ConfiguredCall Returns<T>(MatchArgs matchArgs, T returnThis, par
             }
             else
             {
-                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese));
+                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese).ToArray());
```

Comment:
I prefer to limit the definition to more strict types, array in this case. `IEnumerable<T>` means not-materialized-in-memory/ undefined stream of data. The method taking `IEnumerable<T>` as an argument should take that into account and be lazy all the way down. We have finite and defined number of arguments in our case.